bee-protocol/src/worker/mod.rs

```diff
@@ -39,3 +39,17 @@ pub(crate) use solidifier::{
 pub(crate) use status::StatusWorker;
 pub(crate) use tps::TpsWorker;
 pub(crate) use transaction::{HasherWorker, HasherWorkerEvent, ProcessorWorker};
+
+use bee_common::worker::Error as WorkerError;
+
+use async_trait::async_trait;
+use futures::Stream;
+
+#[async_trait]
+pub(crate) trait Worker {
+    type Event;
+
+    type Receiver: Stream<Item = Self::Event>;
```

Comment:
I don't think that `Event`/`Receiver` should exist here. Doing things through the event bus by passing a reference to the node in might work better here. Something like this:

```rust
impl<N: Node> Worker<N> for MyWorker {
    fn start(node: &N) -> Result<Self, Self::Error> {
        let (tx, rx) = channel();
        node.event_bus().register::<MyEvent>(move |e| tx.send(e.clone()));

        task::spawn(move || {
            while let Ok(e) = tx.recv().await {
                // Use `e` here
            }
        });
    }
}
```